src/RuleSet.php

```diff
@@ -56,12 +56,53 @@ final class RuleSet implements RuleSetInterface
             'switch_case_space' => true,
             'visibility_required' => true,
         ],
-        '@Symfony' => [
+        '@PSR12' => [
```

Comment:
The `braces` fixer should be configured to allow single_line closure, rather than inheriting the defaults from the PSR2 preset.

Reply:
`braces => ['allow_single_line_closure' => true]`

Reply:
We're also missing `'concat_space' => ['spacing' => 'one']`.

Reply:
`'function_typehint_space' => true` and `'no_leading_namespace_whitespace' => true` are also missing and should be included (I think).

Reply:
> The `braces` fixer should be configured to allow single_line closure

I didn't find anything in the spec that dictates this. Could you please point me at the correct place?

> the spec has an example allowing `{}` body for anonymous classes

Added option `allow_single_line_anonymous_class_with_empty_body`.

> We're also missing `'concat_space' => ['spacing' => 'one']`

Actually adding it would be too strict for [the same reason you mentioned for `binary_operator_spaces`](https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4943#discussion_r460592408).

> `'function_typehint_space' => true` and `'no_leading_namespace_whitespace' => true` should be included

I also didn't find anything in the spec for this.

Reply:
> I didn't find anything in the spec that dictates this. Could you please point me at the correct place?

Yeh, I can't see it now. Maybe they took it out of the final thing. I remember it was certainly discussed.

> I also didn't find anything in the spec for this.

`function_typehint_space` is covered by section 4.4 and 4.5. They don't spell it out in text, but they say "Note the placement of parentheses, commas, spaces, and braces". I think they did intend to say it, they just didn't write it down properly.

> no_leading_namespace_whitespace

I think this one is implied by their examples, where they are trying to define the spec by examples, instead of by spelling out the rules. Kinda a shame they did that.

Reply:
I've updated my PR again.

Reply:
Rules inferred from examples might be subject to interpretation. I'm not sure we should add those. @SpacePossum @kubawerlos, what do you think?

Reply:
I totally agree the spec is not clear enough too, and we could even go further as to seek an clarification from the authors and/or a spec amendment. It's a shame they didn't get php-cs-fixer and phpcs authors to help write the spec...

Reply:
I definitely agree, I'm asking myself the same question.